dll/win32/shlwapi/ordinal.c

```diff
@@ -212,25 +213,23 @@ HANDLE WINAPI SHAllocShared(LPCVOID lpvData, DWORD dwSize, DWORD dwProcId)
 LPVOID WINAPI
 SHLockSharedEx(HANDLE hShared, DWORD dwProcId, BOOL bWriteAccess)
 {
-    HANDLE hDup;
-    LPVOID pMapped;
-    DWORD dwAccess;
+  HANDLE hDup;
+  LPVOID pMapped;
+  DWORD dwAccess;
 
-    TRACE("(%p %d %d)\n", hShared, dwProcId, bWriteAccess);
+  TRACE("(%p %d %d)\n", hShared, dwProcId, bWriteAccess);
 
-    /* Get handle to shared memory for current process */
-    hDup = SHMapHandle(hShared, dwProcId, GetCurrentProcessId(), FILE_MAP_ALL_ACCESS, 0);
-    if (!hDup)
```

Comment:
Why remove this check? It looked like an improvement...

Reply:
I would then suggest @katahiromz to make a separate PR / send a separate patch to wine for this.

Reply:
If hDup is NULL, then MapViewOfFile will return NULL.

Reply:
Fixed in 5a440b2.

Reply:
> separate

(Agreed.)

> MapViewOfFile will return NULL

From a Wine point of view, they may prefer less checks.

> Fixed

From a ReactOS point of view, I would suggest to add `if (hShared == NULL) return NULL;` even, assuming that matches Windows result.

Reply:
> 5a440b2

You updated the wrong function.

An additional tweak: `dwProcId` should have `%u`.